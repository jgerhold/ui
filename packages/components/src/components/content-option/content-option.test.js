import { mount } from '@vue/test-utils'
import { selectContent } from '@podlove/player-actions/share'
import ContentOption from './ContentOption'

describe('ContentOption', () => {
  describe('component', () => {
    test('should be a Vue instance', async () => {
      const wrapper = await mount(ContentOption)
      expect(wrapper.isVueInstance()).toBeTruthy()
    })
  })

  describe('props', () => {
    test('should render a default', async () => {
      const wrapper = await mount(ContentOption)
      expect(wrapper.element).toMatchSnapshot()
    })

    const props = [
      {
        prop: 'content',
        value: 'test content'
      },
      {
        prop: 'title',
        value: 'test title'
      },
      {
        prop: 'type',
        value: 'test type'
      },
      {
        prop: 'a11y',
        value: 'test a11y'
      },
      {
        prop: 'background',
        value: '#fff'
      },
      {
        prop: 'color',
        type: '#fff'
      }
    ]

    props.forEach(({ prop, value }) => {
      test(`should render with prop ${prop}`, async () => {
        const wrapper = await mount(ContentOption, {
          propsData: {
            [prop]: value
          }
        })

        expect(wrapper.element).toMatchSnapshot()
      })
    })
  })

  describe('events', () => {
    test('should emit SELECT_CONTENT on click', async () => {
      const wrapper = await mount(ContentOption, {
        propsData: {
          type: 'foobar'
        }
      })

      wrapper.trigger('click')
      expect(wrapper.emitted('click')).toEqual([[selectContent('foobar')]])
    })
  })
})
