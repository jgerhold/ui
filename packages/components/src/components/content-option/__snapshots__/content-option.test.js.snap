// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentOption props should render a default 1`] = `
<div
  aria-label=""
  class="content-option"
  role="button"
  style="background: rgb(43, 138, 198); color: rgb(255, 255, 255);"
>
  <div
    class="content-option-icon"
  />
   
  <!---->
   
  <!---->
</div>
`;

exports[`ContentOption props should render with prop a11y 1`] = `
<div
  aria-label="test a11y"
  class="content-option"
  role="button"
  style="background: rgb(43, 138, 198); color: rgb(255, 255, 255);"
>
  <div
    class="content-option-icon"
  />
   
  <!---->
   
  <!---->
</div>
`;

exports[`ContentOption props should render with prop background 1`] = `
<div
  aria-label=""
  class="content-option"
  role="button"
  style="background: rgb(255, 255, 255); color: rgb(255, 255, 255);"
>
  <div
    class="content-option-icon"
  />
   
  <!---->
   
  <!---->
</div>
`;

exports[`ContentOption props should render with prop color 1`] = `
<div
  aria-label=""
  class="content-option"
  role="button"
  style="background: rgb(43, 138, 198); color: rgb(255, 255, 255);"
>
  <div
    class="content-option-icon"
  />
   
  <!---->
   
  <!---->
</div>
`;

exports[`ContentOption props should render with prop content 1`] = `
<div
  aria-label=""
  class="content-option"
  role="button"
  style="background: rgb(43, 138, 198); color: rgb(255, 255, 255);"
>
  <div
    class="content-option-icon"
  />
   
  <span
    class="content-option-type"
  >
    test content
  </span>
   
  <!---->
</div>
`;

exports[`ContentOption props should render with prop title 1`] = `
<div
  aria-label=""
  class="content-option"
  role="button"
  style="background: rgb(43, 138, 198); color: rgb(255, 255, 255);"
>
  <div
    class="content-option-icon"
  />
   
  <!---->
   
  <span
    class="content-option-title truncate"
  >
    test title
  </span>
</div>
`;

exports[`ContentOption props should render with prop type 1`] = `
<div
  aria-label=""
  class="content-option"
  role="button"
  style="background: rgb(43, 138, 198); color: rgb(255, 255, 255);"
>
  <div
    class="content-option-icon"
  />
   
  <!---->
   
  <!---->
</div>
`;
